pkg/store/consul/auditlogstore/consul_store.go

```diff
@@ -0,0 +1,128 @@
+package auditlogstore
+
+import (
+	"encoding/json"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/square/p2/pkg/util"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/pborman/uuid"
+)
+
+const auditLogTree string = "audit_logs"
+
+// ConsulStore is responsible for storage and retrieval of AuditLog JSON
+// objects. It operates a little differently than the other consul store types
+// in this project because the creation of an audit record is always expected
+// to be a part of a transaction. As a result, it doesn't actually need access
+// to a consul client to create records, it will simply save them to a
+// transaction object.
+type ConsulStore struct {
+	consulKV ConsulKV
+}
+
+type ConsulKV interface {
+	List(prefix string, opts *api.QueryOptions) (api.KVPairs, *api.QueryMeta, error)
+}
+
+func NewConsulStore(consulKV ConsulKV) ConsulStore {
+	return ConsulStore{
+		consulKV: consulKV,
+	}
+}
+
+func (ConsulStore) Create(
+	txn *api.KVTxnOps,
```

Comment:
I think we'll have 3 options:

1) modify the existing functions (e.g. AddDesiredReplicas) to create their own transactions and handle audit logging, maybe the store has a configuration option to turn this off and on.
2) create new functions alongside the old that do this
3) pass in transaction to AddDesiredReplicas() which makes the caller responsible for its audit logs